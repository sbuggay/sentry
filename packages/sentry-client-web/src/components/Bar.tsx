import * as React from "react";

interface IBarProps {
    percentage: number;
    text?: string;
    style?: React.CSSProperties;
    barStyle?: React.CSSProperties;
}

export class Bar extends React.Component<IBarProps, any> {

    public getStyle(): React.CSSProperties {
        return {
            width: "100%",
            height: "16px",
            boxSizing: "border-box",
            border: "1px solid gray",
            borderRadius: "4px",
            ...this.props.style
        };
    }

    public getBarStyle(): React.CSSProperties {
        return {
            width: `${this.props.percentage}%`,
            height: "100%",
            background: "#62C552",
            borderRadius: "4px",
            ...this.props.barStyle
        };
    }

    public getTextStyle(): React.CSSProperties {
        return {
            float: "right",
            fontSize: "12px",
            marginRight: "4px",
            clear: "both",
            lineHeight: this.getStyle().height
        };
    }

    public render(): JSX.Element | null {
        return (
            <div style={this.getStyle()}>
                <div style={this.getTextStyle()}>
                    {this.props.text ? this.props.text : null}
                </div>
                <div style={this.getBarStyle()}>
                </div>
            </div>
        );
    }
}

export default Bar;
