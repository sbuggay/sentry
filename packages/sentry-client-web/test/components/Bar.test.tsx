import * as React from "react";
import * as renderer from "react-test-renderer";

import Bar from "../../src/components/Bar";

describe("Bar", () => {

    it("renders a basic Bar", () => {
        const component = renderer.create(
            <Bar percentage={50} />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});
