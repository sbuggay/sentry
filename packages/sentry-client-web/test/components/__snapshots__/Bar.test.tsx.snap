// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar renders a basic Bar 1`] = `
<div
  style={
    Object {
      "border": "1px solid gray",
      "borderRadius": "4px",
      "boxSizing": "border-box",
      "height": "16px",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "clear": "both",
        "float": "right",
        "fontSize": "12px",
        "lineHeight": "16px",
        "marginRight": "4px",
      }
    }
  />
  <div
    style={
      Object {
        "background": "#62C552",
        "borderRadius": "4px",
        "height": "100%",
        "width": "50%",
      }
    }
  />
</div>
`;
